### DISCUSSION TITLE
(code review) example of how to create a reusable store with computed/derived properties? is it worth it with zustand?

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [pmndrs/zustand#2077] [Q&A]
Here you can see an example of a reusable timer with computed/derived properties con zustand.

https://codesandbox.io/p/github/MiniSuperDev/reusable-timer-zustand/main?file=%2Fsrc%2Ftimer-store.ts%3A3%2C1
https://github.com/MiniSuperDev/reusable-timer-zustand/tree/main

I don't know if it's the best way, so I want you to please tell me what the best way to do it would be.
 
My idea is to create some UI Framework agnostic reusable stores so I can reuse them in any framework, like react, cli, vue, etc.

I want to see how this would be possible with zustand and if you could tell me if it is worth it or if there is another alternative.

The idea is to be able to create new instances of each store, and in the app I decide if it will be or global o local using the corresponding provider or injector or whatever corresponds.

Thank you so much

**Comment by dbritto-dev** [comment]
@MiniSuperDev I would use finite state machines to build yo components if that'a possible -> here you go https://github.com/chakra-ui/zag
